subsys/net/lib/azure_iot_hub/CMakeLists.txt

```diff
@@ -4,10 +4,42 @@
 # SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
 #
 
+set(azure_sdk_for_c_SOURCE_DIR $ENV{ZEPHYR_BASE}/../modules/lib/azure-sdk-for-c)
+
 zephyr_library()
+
+# Turn off precondition checking and logging to reduce the code footprint and RAM usage
+add_compile_definitions(AZ_NO_PRECONDITION_CHECKING)
+add_compile_definitions(AZ_NO_LOGGING)
```

Comment:
Shouldn't this be configurable with Kconfig?

Reply:
Ill add this to the backlog as it may require some extra work. If they are not enabled it causes the application to hault after a while which has proven a little tricky to debug.